Amethyst/Managers/FocusFollowsMouseManager.swift

```diff
@@ -14,21 +14,65 @@ protocol FocusFollowsMouseManagerDelegate: class {
     func windowsForFocusFollowsMouse() -> [SIWindow]
 }
 
-final class FocusFollowsMouseManager {
+// need to inherit from NSObject to track UserDefaults changes
+final class FocusFollowsMouseManager: NSObject {
     weak var delegate: FocusFollowsMouseManagerDelegate?
 
     private let userConfiguration: UserConfiguration
+    private var focusConfigurationChangeHandler: (Bool) -> Void   // var not let.  necessary for closure setup.
     private var mouseMovedEventHandler: Any?
 
     init(userConfiguration: UserConfiguration) {
         self.userConfiguration = userConfiguration
-        mouseMovedEventHandler = NSEvent.addGlobalMonitorForEvents(matching: .mouseMoved) { event in
-            self.focusWindowWithMouseMovedEvent(event)
+        self.mouseMovedEventHandler = nil
+
+        // must fully init self before we can use it in a closure, so we put in a fake closure for the handler (for now)
+        self.focusConfigurationChangeHandler = { _ in () }
+        super.init()  // fully init self
+
+        // can now set up self as an observer.
+        // we want to observe changes to the focusFollowsMouse config, because mouse tracking has CPU cost
+        UserDefaults.standard.addObserver(self, forKeyPath: ConfigurationKey.focusFollowsMouse.rawValue, options: NSKeyValueObservingOptions.new, context: nil)
```

Comment:
This should probably be behind user configuration. Also, take a look at RxCocoa bindings to `UserDefaults`. It's a bit of a different look at it, but you can do something like
```swift
UserDefaults.standard.rx.observe(...)
    .distinctUntilChanged()
    .scan(nil) { monitor, followingIsDesired in
        if let monitor = monitor {
            NSEvent.removeMonitor(monitor)
            return nil
        } else if followingIsDesired {
            return NSEvent.addGlobalMonitorForEvents(...)
        } else {
            return nil
        }
    }
    .subscribe(...)
```
It takes a bit of grokking the first few times looking at it like this, but it avoids some of the state tracking necessary and is a bit easier to read.